iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift

```diff
@@ -76,6 +63,57 @@ class ConversationCollectionViewDataSource: CollectionViewDataSource<Conversatio
         return nil
     }
 
+    /// Updates the datasource with the passed in array of message changes.
+    func updateMessages(with changes: [ListChange<ChatMessage>],
+                        conversationController: ChatChannelController,
+                        collectionView: UICollectionView) async {
```

Comment:
Moving this update message logic into the datasource to declutter the conversation view controller.